### PR TITLE
Auth reimpl

### LINKED ISSUE #1084: Auth updates
The auth integration is changing slightly and will require updates. More details coming.

### PR DESCRIPTION
**Pull request by wpears**
Closes #1084 

Reimplements most of the auth (minus user info bit). Built on the keycloak adapter rather than oidc-client.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,15 +16,19 @@
...
+    "HMDA"
+  ],
   "author": "Consumer Financial Protection Bureau",
   "license": "CC0-1.0",
   "bugs": {
     "url": "https://github.com/cfpb/hmda-platform-ui/issues"
   },
   "dependencies": {
     "detect-browser": "1.7.0",
+    "eslint-plugin-prettier": "3.0.0",
```
Comment (awolfe76) [inline package.json:29]: Do we need this dependency?

**Review by awolfe76 — CHANGES_REQUESTED** [review]
It looks like router is catching the `/auth`. I'm seeing 
`Warning: [react-router] Location "/auth/realms/hmda2/protocol/openid-connect/auth?client_id=hmda2-api&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Ffiling%2Finstitutions&state=64a50230-645d-4020-8577-595fd81fe242&response_mode=fragment&response_type=code&scope=openid&nonce=5a0cd74b-31ee-4a16-9bd4-63817ed9f00d" did not match any routes`.

Maybe a different setup? I'm using dev for the proxy.

_Also pushed up 5f0aa6f, just a quick clean up._

**Comment by awolfe76** [comment]
Couldn't we also go ahead and add the [`Authorization` header to the API requests](https://github.com/cfpb/hmda-platform-ui/blob/0103791272c8366c163ef9b03ffc16b13ebaed88/src/api/fetch.js#L56-L60)?

File: package.json (hunk 1 above)
Comment (wpears) [reply package.json:29]: Without it, CRA breaks my local tooling :sweat:

**Review by awolfe76 — APPROVED** [review]

**Review by awolfe76 — APPROVED** [review]
It looks like router is catching the `/auth`. I'm seeing 
`Warning: [react-router] Location "/auth/realms/hmda2/protocol/openid-connect/auth?client_id=hmda2-api&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Ffiling%2Finstitutions&state=64a50230-645d-4020-8577-595fd81fe242&response_mode=fragment&response_type=code&scope=openid&nonce=5a0cd74b-31ee-4a16-9bd4-63817ed9f00d" did not match any routes`.

Maybe a different setup? I'm using dev for the proxy.

**Pull request merged**